pageserver/src/page_service.rs

```diff
@@ -245,12 +248,14 @@ async fn page_service_conn_main(
         .set_nodelay(true)
         .context("could not set TCP_NODELAY")?;
 
+    let peer_addr = socket.peer_addr().context("get peer address")?;
+
     // XXX: pgbackend.run() should take the connection_ctx,
     // and create a child per-query context when it invokes process_query.
     // But it's in a shared crate, so, we store connection_ctx inside PageServerHandler
     // and create the per-query context in process_query ourselves.
     let mut conn_handler = PageServerHandler::new(conf, auth, connection_ctx);
-    let pgbackend = PostgresBackend::new(socket, auth_type, None)?;
+    let pgbackend = PostgresBackend::new_from_io(socket, peer_addr, auth_type, None)?;
```

Comment:
This change seems unnecessary. `PostgresBackend::new` would still work here, right?

Reply:
Agreed, most likely. Do you want me to split this?

Reply:
Up to you.

Does the next PR need this change either?

Reply:
Yes it does, because the stream will no longer be plain `tokio::net::TcpStream` but `tokio_io_util::TimeoutRead<tokio::net::TcpStream>`.

I think this is good enough.